angular/demo/src/app/samples/pagination/default.route.ts

```diff
@@ -6,23 +6,46 @@ import {Component} from '@angular/core';
 	imports: [AgnosUIAngularModule],
 	template: `
 		<h5>Basic pagination:</h5>
-		<nav auPagination auCollectionSize="60" [(auPage)]="page"></nav>
+		<nav auPagination auCollectionSize="60" [(auPage)]="page" [auPageLink]="pageLink"></nav>
 
 		<h5>No direction links:</h5>
-		<nav auPagination auCollectionSize="60" [(auPage)]="page" [auDirectionLinks]="false"></nav>
+		<nav auPagination auCollectionSize="60" [(auPage)]="page" [auPageLink]="pageLink" [auDirectionLinks]="false"></nav>
 
 		<h5>With boundary links:</h5>
-		<nav auPagination auCollectionSize="60" [(auPage)]="page" auBoundaryLinks></nav>
+		<nav auPagination auCollectionSize="60" [(auPage)]="page" [auPageLink]="pageLink" auBoundaryLinks></nav>
 
 		<div class="mb-3">
 			Current page: <span id="defaultPage">{{ page }}</span>
 		</div>
 
 		<h5>Disabled pagination:</h5>
-		<nav auPagination auCollectionSize="60" [(auPage)]="pageAlone" auAriaLabel="Disabled pagination" auDisabled></nav>
+		<nav auPagination auCollectionSize="60" [(auPage)]="pageAlone" [auPageLink]="pageLink" auAriaLabel="Disabled pagination" auDisabled></nav>
 	`,
 })
 export default class DefaultPaginationComponent {
 	page = 4;
 	pageAlone = 1;
+
+	constructor() {
+		let jsonObj: {page?: number} = {};
```

Comment:
Is it possible to export this logic in an angular service ?

Reply:
I could do it in an angular service, or I could adapt [this function](https://github.com/AmadeusITGroup/AgnosUI/blob/2a92391c534dde7356367de324b8f246e1811f51/common/utils.ts#L17)
 to return the full JSON object, and use it in the other apps in the example.
 
BTW I was thinking to do another PR for the example ( and remove it from this one just before merge).
what do you think ?

Reply:
As it's a simple example, maybe you can simplify the way it's managed with the hash. For example, you can just manage an fragment like this :

`#${hashFragment}#${_pageNumber}}`

It would simplify the logic of the example.

By the way, it's no easy for the user to know what's happened here, because we are changing (of course) the url of the iframe, but it's not directly visible. I don't know how it have it clearer... (specific example where the hash is output on the display ?)

Reply:
I don't know about the function. maybe we need to discuss if this is interesting to share it. If not service. Bring this into a meeting.

Reply:
After discussions the examples have been a lot simplified. There is a small common part which exposes the hash as a store.